resources/common.js

```diff
@@ -80,3 +80,154 @@ let Util = {
     return kAddleJobs.indexOf(job) >= 0;
   },
 };
+
+(function() {
+  let wsUrl = /[\?&]OVERLAY_WS=([^&]+)/.exec(location.href);
+  let ws = null;
+  let queue = [];
+  let rseqCounter = 0;
+  let responsePromises = {};
+  let subscribers = {};
+  let sendMessage = null;
+
+  if (wsUrl) {
+    sendMessage = (obj) => {
+      if (queue)
+        queue.push(msg);
```

Comment:
I think you mean obj here? (or msg to match what's below)?

Even changing that, I am still unable to get websocket support to work properly with cactbot.  If I start the server (with ssl), I can see the "It works!" page.  Then if I go to `file:///C:/Users/enne/cactbot/ui/jobs/jobs.html?OVERLAY_WS=ws://127.0.0.1:10501` I get this error:
```
common.js:102 WebSocket connection to 'ws://127.0.0.1:10501/' failed: Error during WebSocket handshake: net::ERR_CONNECTION_RESET
```

Reply:
Oh, uh.... yes, that should've been `obj`. Can't believe I missed that.

Anyway, the `ERR_CONNECTION_RESET` is caused by using a `ws://` URL with a SSL enabled server. If you have SSL enabled, you need `wss://`. It's just like `http://` and `https://`.

Reply:
Sorry for ongoing support questions, but if I do that, then I get:
```
common.js:102 WebSocket connection to 'wss://127.0.0.1:10501/' failed: Error during WebSocket handshake: Unexpected response code: 501
```

Reply:
Sorry, I forgot that the URL is supposed to be `file:///C:/Users/enne/cactbot/ui/jobs/jobs.html?OVERLAY_WS=wss://127.0.0.1:10501/ws`. I really need to write this down sometime.

The currently supported endpoints are:
* `/ws`: This is the "new" API and allows you to do the same things that are possible in an overlay.
* `/MiniParse`: This endpoint yields `CombatData` events but ignores all input. This was implemented to be backwards compatible with ACTWebSocket. This is enough to support MopiMopi and Kagerou.
* `/BeforeLogLineRead`: This endpoint currently does the same as `/MiniParse` which means that it's still missing data. I started implementing this to support SkillDisplay though I haven't finished it, yet.

Reply:
AHA! :confetti_ball: Thanks!